src/scanners/css.js

```diff
@@ -52,56 +52,49 @@ export default class CSSScanner extends BaseScanner {
       _rules[cssInstruction](cssNode, file, cssOptions));
   }
 
-  scan(_rules = this._defaultRules) {
-    return new Promise((resolve, reject) => {
-      this.getContents()
-        .then((ast) => {
-          if (ast && ast.nodes) {
-            const rules = ignorePrivateFunctions(_rules);
-            const { nodes } = ast;
-
-            Object.keys(rules).forEach((cssInstruction) => {
-              this._rulesProcessed++;
-              nodes.forEach((cssNode) => {
-                this.processCode(cssNode, cssInstruction, rules);
-              });
-            });
-          }
-
-          resolve({
-            linterMessages: this.linterMessages,
-            scannedFiles: this.scannedFiles,
-          });
-        })
-        .catch(reject);
-    });
+  async scan(_rules = this._defaultRules) {
+    const ast = await this.getContents();
+    if (ast && ast.nodes) {
+      const rules = ignorePrivateFunctions(_rules);
+      const { nodes } = ast;
+
+      Object.keys(rules).forEach((cssInstruction) => {
+        this._rulesProcessed++;
+        nodes.forEach((cssNode) => {
+          this.processCode(cssNode, cssInstruction, rules);
+        });
+      });
+    }
+
+    return {
+      linterMessages: this.linterMessages,
+      scannedFiles: this.scannedFiles,
+    };
   }
 
-  _getContents(_cssParser = postcss) {
-    return new Promise((resolve, reject) => {
-      try {
-        const rootNode = _cssParser.parse(this.contents, { from: this.filename });
-        return resolve(rootNode);
-      } catch (e) {
-        if (!e.reason || e.name !== 'CssSyntaxError') {
-          return reject(e);
-        }
-        this.linterMessages.push(Object.assign({}, CSS_SYNTAX_ERROR, {
-          type: VALIDATION_WARNING,
-          // Use the reason for the error as the message.
-          // e.message includes an absolute path.
-          message: e.reason,
-          column: e.column,
-          line: e.line,
-          // We use our own ref to the file as postcss outputs
-          // absolute paths.
-          file: this.filename,
-        }));
-
-
-        // A syntax error has been encounted so it's game over.
-        return resolve(null);
+  async _getContents(_cssParser = postcss) {
+    try {
+      const rootNode = _cssParser.parse(this.contents, { from: this.filename });
```

Comment:
`rootNode` doesn't seem to be used anywhere in this function, we could just return the value returned by `_cssParser.parse`.